File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/reporting/PartnerChargeMethodSales.ts
import { ChargePartner } from "../acquirerconfigurations/ChargePartner";
import { ChargeMethod } from "../ChargeMethod";

export interface PartnerChargeMethodSales {
    readonly chargePartner: ChargePartner;
    readonly chargeMethod: ChargeMethod;

    readonly from: string;
    readonly to: string;

    readonly total: number;
    readonly totalSuccess: number;
    readonly totalFailed: number;
    readonly totalProcessing: number;
}

File: Quivi.Backoffice.Api/Dtos/PartnerChargeMethodSales.cs
using Quivi.Domain.Entities.Charges;

namespace Quivi.Backoffice.Api.Dtos
{
    public class PartnerChargeMethodSales
    {
        public required ChargePartner ChargePartner { get; init; }
        public required ChargeMethod ChargeMethod { get; init; }

        public required DateTimeOffset From { get; init; }
        public required DateTimeOffset To { get; init; }

        public int Total { get; init; }
        public int TotalSuccess { get; init; }
        public int TotalFailed { get; init; }
        public int TotalProcessing { get; init; }
    }
}